.github/workflows/main.yml

```diff
@@ -17,6 +17,9 @@ jobs:
           - 'registry.fedoraproject.org/fedora:rawhide'
           - 'registry.opensuse.org/opensuse/leap-dnf:15.3'
           - 'registry.opensuse.org/opensuse/tumbleweed-dnf:latest'
+        build-type:
+          - 'normal'
+          - 'no-optional-deps'
```

Comment:
Nice! I would not run all 2x4 combinations, but I would rather use `include` and run only 2 `no-optional-deps` (one fedora and one opensuse).